feat(tabs): add bottom tab navigator with Home, Library, Calendar, Ch…

ThanhDatVu111 opened the pull request:
## 🚀 What does this PR do?

>- Adds a persistent bottom tab navigator with five tabs—Home, Library, Calendar, Chat, and Profile—using Expo Router’s advanced tabs.
>- Introduces TabsLayout under app/(tabs)/_layout.tsx
>- Registers each screen with a custom TabIcon that switches between focused (highlighted background + label) and unfocused states
>- Applies rounded styling and hides default labels via tabBarStyle and tabBarItemStyle

---

## ✅ Checklist

- [x] Code compiles & runs
- [x] Lint checks pass
- [ ] Updated `.env.example` if needed
- [x] PR linked to related ticket #38

---

## 🔍 How to test this

> Please cd into frontend and then sign in to test the bottom bar

<img width="429" alt="Screenshot 2025-05-04 at 9 41 07 PM" src="https://github.com/user-attachments/assets/aa7c3750-ebd1-4930-bb57-d752d5d24ca9" />

---

## 🧠 Extra Notes

> If you add or rename a tab file, update the <Tabs.Screen name="…"> entries to match exactly.

ThanhDatVu111 committed:
feat(tabs): add bottom tab navigator with Home, Library, Calendar, Chat, and Profile

tranguv commented on frontend/src/app/(tabs)/_layout.tsx:
```diff
@@ -0,0 +1,105 @@
+import images from '@/constants/images';
+import { ImageBackground, Image, Text, View } from 'react-native';
+
+function TabIcon({ focused, icon, title }: any) {
```
Fix any type to TabIconProps like this

interface TabIconProps {
  focused: boolean;
  icon: ImageSourcePropType;
  title: string;
}

ThanhDatVu111 replied:
Done

tranguv commented on frontend/src/app/(tabs)/home.tsx:
```diff
@@ -0,0 +1,12 @@
+import { View, Text } from "react-native";
+import React from "react";
+
+const home = () => {
```
use uppercase (Home) for consistency naming convention maybe (?)

ThanhDatVu111 replied:
I have change everyfile

tranguv commented on frontend/src/app/(tabs)/_layout.tsx:
```diff
@@ -0,0 +1,105 @@
+          position: 'absolute',
+          overflow: 'hidden',
+          borderWidth: 1,
+          borderColor: '',
```
try removing borderColor: ' ' and add drop shadow instead

ThanhDatVu111 replied:
Done

tranguv reviewed (changes requested):
I think the revised bottom navigation bar is lovely. Good job! Just some concerns I have noticed:
1. There are some UI issues that you may have noticed
<img width="340" alt="image" src="https://github.com/user-attachments/assets/37618da4-6c04-4693-9cc6-aba4b95f310c" />
<img width="342" alt="Screenshot 2025-05-13 at 10 26 51 PM" src="https://github.com/user-attachments/assets/d300c52d-a573-4dd0-8ad4-ecd568720e6a" />

2. I love how you have the purple highlighted oval when users switch tabs, but with five tabs, do you think it feels a bit crowded...? I think 4 is still acceptable but 5 is a bit too much
<img width="122" alt="Screenshot 2025-05-13 at 10 32 29 PM" src="https://github.com/user-attachments/assets/0b383aa9-4029-41e2-a3b8-8f8ca215a197" />

3. It would be great if you include Android UI as well. You can see the purple oval is not aligned well within the container
<img src="https://github.com/user-attachments/assets/2447fa6b-3ec0-4c03-8193-d984a1b305f3" alt="Image" width="200" height="auto">

tranguv commented on frontend/src/app/(tabs)/chat.tsx:
```diff
@@ -0,0 +1,12 @@
+
+const chat = () => {
+  return (
+    <View className = "flex-1 justify-center items-center bg-primary">
```
I see this styling is repetitive across multiple files, perhaps, we can create a tailwind css rule for it...?

ThanhDatVu111 replied:
I tried to put shortcut inside tailwindconfig.js and call the shortcut inside tabs file , but it never worked. dont really know why. I will try to seek help from office hour!

tranguv replied:
Try these suggestions from [Stack](https://stackoverflow.com/questions/78535645/after-setup-nativewind-does-not-work-with-expo) and let me know if it works

ThanhDatVu111 replied:
Hi ma'am, I used a CSS plugin to fix it in my latest commit today. Sorry for the delay!

ThanhDatVu111 committed:
fixed  styling for tabs and added some minor changes

ThanhDatVu111 committed:
Add tab-screen Tailwind plugin, bump Expo versions & refactor screens

- Introduce a custom  component via TailwindCSS plugin
- Upgrade Expo SDK dependencies (expo, expo-router, expo-splash-screen) to expected patch versions
- Refactor Chat, Calendar, Message & Home screens to use the new shared CSS utility for cleaner markup

ThanhDatVu111 committed:
change package json lock in backend

tranguv reviewed (approved):
Thanks for fixing this! LGTM

MC2114 reviewed (approved):
I think the Nav Bar looks good for now! We can update it in the future if we want to change the look a bit but I think it is okay (since we don't need the `+` button anymore)
